sqlserver/tests/test_statements.py

```diff
@@ -311,10 +314,12 @@ def test_statement_metrics_and_plans(
     # 2) load the test queries into the StatementMetrics state
     # 3) emit the query metrics based on the diff of current and last state
     dd_run_check(check)
+    time.sleep(2)
```

Comment:
tag sleep

Reply:
why do we need sleep here? I thought we need either sleep or a close to zero collection interval but not both

Reply:
if `sleep` is needed regardless of the collection interval, can we just add sleep without setting the collection interval to 0.000001 to minimize the changes?

Reply:
Adding a bunch of sleeps all over the place seems like a recipe for disaster. It also feels like setting a very low collection interval is hacky / prone to future flakey-ness / not straightforward _why_ we need it.  I would just add a `run_sync_blocking` function or something like that to the DBMAsync job, which has the old behavior and then update the tests to use that. This is a bit weird, because we are adding production code that will only be used by tests, but we do this in plenty of places in the backend and this seems like a fair trade off in order to have reliable tests going forward. We just need to make sure this function is documented well.